Docs: Updates docs for redux framework (#20377)
grafana/grafana: contribute/style-guides/redux.md

  Object literal may only specify known properties, and 'nonExistingProperty' does not exist in type 'ExampleReducerState'. [2322]
```

## Test functionality

### reducerTester

Fluent API that simplifies the testing of reducers

#### Usage

```typescript
reducerTester()
  .givenReducer(someReducer, initialState)
  .whenActionIsDispatched(someAction('reducer tests'))
  .thenStateShouldEqual({ ...initialState, data: 'reducer tests' });
```

#### Complex usage
Sometimes you encounter a `resulting state` that contains properties that are hard to compare, such as `Dates`, but you still want to compare that other props in state are correct.

> Then you can use `thenStatePredicateShouldEqual` function on `reducerTester` that will return the `resulting state` thatyou can expect upon.

Reviewer: s/thatyou/that you/

Reviewer: ...expect upon individual properties.

Author: Thanks David!

  Object literal may only specify known properties, and 'nonExistingProperty' does not exist in type 'ExampleReducerState'. [2322]
```

## Test functionality

### reducerTester

Fluent API that simplifies the testing of reducers

> #### Simple Example

Reviewer 2: ```suggestion
#### Simple example
```

[Write headings in sentence case, not title case](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md#headings)

> #### Simple Example

Reviewer 3: ```suggestion
#### Usage
```

Author: I like this better, thanks

Reviewer 3: I would go with `Usage` everywhere when `[Simple|Complex] example` is used

```typescript
reducerTester()
  .givenReducer(someReducer, initialState)
  .whenActionIsDispatched(someAction('reducer tests'))
  .thenStateShouldEqual({ ...initialState, data: 'reducer tests' });
```

> #### Complex Example

Reviewer 2: ```suggestion
#### Complex example
```

[Write headings in sentence case, not title case](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md#headings)

> Sometimes you encounter a `resulting state` that contains properties that are hard to compare like `Dates` but you still 

Reviewer 2: ```suggestion
Sometimes you encounter a `resulting state` that contains properties that are hard to compare, such as `Dates`, but you still 
```
"compare like dates" sounds like you want to compare the properties _as_ Dates, which I don't think is the intention here?

> want to compare that other props in state are correct.

Reviewer 2: This line seems to be hard wrapped. It should be on the same line as the one above.

Author: Will fix that!

Then you can use `thenStatePredicateShouldEqual` function on `reducerTester` that will return the `resulting state` that

> you can expect upon.

Reviewer 2: This line seems to be hard wrapped. It should be on the same line as the one above.

Author: Will fix that!

```typescript
reducerTester()
  .givenReducer(someReducer, initialState)
  .whenActionIsDispatched(someAction('reducer tests'))
  .thenStatePredicateShouldEqual(resultingState => {
    expect(resultingState.data).toEqual('reducer tests');
    return true;  
  });
```

### thunkTester

> Fluent API that simplifies the testing of thunks

Reviewer 2: ```suggestion
Fluent API that simplifies the testing of thunks.
```
